jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/IOResources.java

```diff
@@ -58,23 +58,20 @@ public static RetainableByteBuffer toRetainableByteBuffer(Resource resource, Byt
             return RetainableByteBuffer.wrap(ByteBuffer.wrap(memoryResource.getBytes()));
 
         long longLength = resource.length();
-        if (longLength > Integer.MAX_VALUE)
-            throw new IllegalArgumentException("Resource length exceeds 2 GiB: " + resource);
-        int length = (int)longLength;
 
-        bufferPool = bufferPool == null ? ByteBufferPool.NON_POOLING : bufferPool;
+        bufferPool = bufferPool == null ? new ByteBufferPool.NonPooling() : bufferPool;
 
         // Optimize for PathResource.
         Path path = resource.getPath();
-        if (path != null)
+        if (path != null && longLength < Integer.MAX_VALUE)
```

Comment:
Wouldn't using a `RetainableByteBuffer.DynamicCapacity` like in the `InputStream` fallback work too?

Reply:
Actually, we need to decide what we want here.   We know the length, so we can read the content into a single byte buffer (FixedCapacity), which is unlikely to be a standard size, but if it is large, it might be better to read it into a DynamicCapacity so that it is stored as a chain of standard sized buffers and thus are better reused.  

However, currently HttpContent is written as a singe getByteBuffer write, so the FixedCapacity approach is better.   WHEN we change HttpContent to use a writeTo semantic, then it will be better to be dynamic here.  I'll add TODOs